src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UI/InfoBars/VsInfoBarService.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.UI.InfoBars;
+
+/// <summary>
+/// Implementation of <see cref="IInfoBarService"/> that pushes messages to the info bar attached to Visual Studio's main window.
+/// </summary>
+[Export(typeof(IInfoBarService))]
+internal sealed class VsInfoBarService : IInfoBarService
+{
+    // This component exists per configured project. We track entries at the global level here, and keep track of the
+    // projects related to each info bar within the entry. This allows us to close info bars when all projects that
+    // requested them are unloaded.
+    private static readonly List<InfoBar> s_entries = [];
+
+    private static IVsInfoBarHost? s_host;
+
+    private readonly UnconfiguredProject _project;
+    private readonly IProjectThreadingService _threadingService;
+    private readonly IVsShellServices _vsShellServices;
+    private readonly IVsUIService<SVsShell, IVsShell> _vsShell;
+    private readonly IVsUIService<SVsInfoBarUIFactory, IVsInfoBarUIFactory> _vsInfoBarFactory;
+
+    [ImportingConstructor]
+    public VsInfoBarService(
+        UnconfiguredProject project,
```

Comment:
@LittleLittleCloud this is the actual change that moves the scope of the MEF component into UnconfiguredProject scope. The previous service only imported from the Glboal and ProjectServices scopes. Now we import UnconfiguredProject, so we're moved into that scope.